internal/pkg/mount/mount.go

```diff
@@ -28,21 +28,31 @@ type RetryFunc func(*Point) error
 
 // Mount mounts the device(s).
 func Mount(mountpoints *Points) (err error) {
+	var mounted bool
+
 	iter := mountpoints.Iter()
 
 	//  Mount the device(s).
 
 	for iter.Next() {
 		mountpoint := iter.Value()
+
 		// Repair the disk's partition table.
 		if mountpoint.Resize {
 			if err = mountpoint.ResizePartition(); err != nil {
 				return fmt.Errorf("error resizing %q: %w", iter.Value().Source(), err)
 			}
 		}
 
-		if err = mountpoint.Mount(); err != nil {
-			return fmt.Errorf("error mounting %q: %w", iter.Value().Source(), err)
+		mounted, err = mountpoint.IsMounted()
```

Comment:
The below `mountpoint.Mount()` _might_ catch this, but its behaviour seems dangerous to me; thus, I chose to always gatekeep the mount with this check.  Intentionally outside this check are the resizing above an below, so as to minimize possible side-effects.